### PR TITLE
fix: add check for TBD datetimes breaking add to calendar button (#66)

### LINKED ISSUE #66: Add to Google Calendar button sets event time to current time instead of event time

### PR DESCRIPTION
**Pull request by ericswpark**
Fixes #66

This needs some review; I was unable to test it because we have no upcoming events and I don't have access to the API.

### CODE REVIEW

File: lib/fetchEvents.ts [hunk 1]
```diff
@@ -11,24 +11,43 @@ const client = createClient({
   useCdn: true,
 })
 
+
+const getSanitizedTime = (start: string, end: string) => {
+  if (start !== "TBD" && end !== "TBD") {
+    return [new Date(start), new Date(end)];
+  } else if (start !== "TBD") {
+    // End hour is undefined, bodge to 1 hour from start time for now
+    var start_date = new Date(start);
```
Comment (MatthewStanciu) [inline lib/fetchEvents.ts:20]:
```suggestion
    let startDate = new Date(start);
```

**Review by MatthewStanciu — COMMENTED** [review]
Nit: prefer camelCase and let/const instead of var

File: lib/fetchEvents.ts (hunk 1 above)
Comment (ericswpark) [reply lib/fetchEvents.ts]: Fixed in styling commit!

**Comment by ericswpark** [comment]
This is much more difficult than I thought. The problem is, if the server and client have different timezones set, then it causes a hydration error on load because the `Date` object constructed server-side doesn't match client-side.

There are ways to fix this -- have the date object be constructed within a `useEffect()`, but then it causes some weird behavior on load. `suppressHydrationWarning` is an option, but personally I don't like it because it feels like hiding the root cause. There should be a deterministic way of converting the API responses into a timezone-specific `Date` object that matches on server and client. Maybe [Moment Timezone](https://momentjs.com/timezone/)?

For now I've pushed up the commits to fix the RangeError on site launch. This PR is nowhere near merge in this state, so will convert to draft.

**Comment by ericswpark** [comment]
Okay, never mind. I probably should've checked if the hydration errors occur on the main branch before digging a rabbit hole for myself, because the current codebase gives you that error regardless of date/time construction. I think this PR is ready to be merged, and future PRs should fix the hydration error.

**Comment by MatthewStanciu** [comment]
STUNNING 🌟

**Pull request merged**